React/Base/RCTTVRemoteHandler.m

```diff
@@ -0,0 +1,160 @@
+                                    pressType:UIPressTypeRightArrow];
+    
+    
+    // Recognizers for Apple TV remote trackpad swipes
+    
+    // Up
+    [self addSwipeGestureRecognizerWithSelector:@selector(swipedUp:)
+                                      direction:UISwipeGestureRecognizerDirectionUp];
+    
+    // Down
+    [self addSwipeGestureRecognizerWithSelector:@selector(swipedDown:)
+                                      direction:UISwipeGestureRecognizerDirectionDown];
+    
+    // Left
+    [self addSwipeGestureRecognizerWithSelector:@selector(swipedLeft:)
+                                      direction:UISwipeGestureRecognizerDirectionLeft];
+    
+    // Right
+    [self addSwipeGestureRecognizerWithSelector:@selector(swipedRight:)
+                                      direction:UISwipeGestureRecognizerDirectionRight];
+    
+  }
+
+  return self;
+}
+
+RCT_NOT_IMPLEMENTED(- (instancetype)init)
+
+
+- (void)playPausePressed:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"playPause" toView:r.view];
+}
+
+- (void)menuPressed:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"menu" toView:r.view];
+}
+
+- (void)selectPressed:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"select" toView:r.view];
+}
+
+- (void)longPress:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"longPress" toView:r.view];
+}
+
+- (void)swipedUp:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"up" toView:r.view];
+}
+
+- (void)swipedDown:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"down" toView:r.view];
+}
+
+- (void)swipedLeft:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"left" toView:r.view];
+}
+
+- (void)swipedRight:(UIGestureRecognizer*)r {
+  [self sendAppleTVEvent:@"right" toView:r.view];
+}
+
+#pragma mark -
+
+- (void)addTapGestureRecognizerWithSelector:(nonnull SEL)selector pressType:(UIPressType)pressType {
+  
+  UITapGestureRecognizer *recognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:selector];
+  recognizer.allowedPressTypes = @[[NSNumber numberWithInteger:pressType]];
+  
+  
+  NSMutableArray *gestureRecognizers = (NSMutableArray*)self.tvRemoteGestureRecognizers;
+  [gestureRecognizers addObject:recognizer];
+}
+
+- (void)addSwipeGestureRecognizerWithSelector:(nonnull SEL)selector direction:(UISwipeGestureRecognizerDirection)direction {
+  
+  UISwipeGestureRecognizer *recognizer = [[UISwipeGestureRecognizer alloc] initWithTarget:self action:selector];
+  recognizer.direction = direction;
+  
+  NSMutableArray *gestureRecognizers = (NSMutableArray*)self.tvRemoteGestureRecognizers;
+  [gestureRecognizers addObject:recognizer];
+}
+
+- (void)sendAppleTVEvent:(NSString*)eventType toView:(UIView*)v {
+  if([v respondsToSelector:@selector(onTVNavEvent)]) {
+    RCTDirectEventBlock onTVNavEvent = [v performSelector:@selector(onTVNavEvent) withObject:nil];
+    if(onTVNavEvent) {
+      onTVNavEvent(@{@"eventType":eventType});
+    }
+  }
+  for(UIView *u in [v subviews]) {
```

Comment:
Keep in mind that TV nav events are rare, and that typically only one or two views would actually implement the onTVNavEvent handler.  So I think this will scale ok.